docs/generic_results.md

```diff
@@ -0,0 +1,64 @@
+
+### ArgusGenericResultMetadata
+This table stores information what results are available for given `test_id` (Jenkins job). Contain information about each result: name, description, columns, and rows.
+
+### ArgusGenericResultData
+This table stores actual data for each result. Each row in this table is a cell in the result table. Its location is defined by `column` and `row` fields.
+Besides the value, each cell contains information about value status (PASS, WARNING, ERROR) and `sut_timestamp` which corresponds to SUT version (e.g. calculated from binaries build date for ScyllaDB). SUT version is not shown in the result table, but it is used for sorting purposes (needed when graphing results in time - time refers to SUT version).
+
+## Usage
+Generic results are stored in Argus database and can be accessed via Argus API and displayed in Argus UI in `Results` tab for given run.
+
+To send generic results to Argus, you need to create `ResultTable` class basing on `GenericResultTable` available in Argus Client library and send it using `ArgusClient` class (or it's child class like `ArgusSCTClient`). 
+
+Child class of `GenericResultTable` needs to define `Meta` class with `name` and `description` fields. It also needs to define `Columns` field that will describe available columns details for given result: as a list of `ColumnMetadata` objects. Each `ColumnMetadata` object needs to have `name`, `unit` and `type` fields. `type` field can be one of `ResultType` enum values.
+
+See example:
+
+```python
+from uuid import UUID
+
+from argus.client.sct.client import ArgusSCTClient
+from argus.client.generic_result import GenericResultTable, ColumnMetadata, ResultType, Status
+
+
+class LatencyResultTable(GenericResultTable):
+    class Meta:
+        name = "Write Latency"
+        description = "Latency for write operations"
+        Columns = [ColumnMetadata(name="latency", unit="ms", type=ResultType.FLOAT),
+                   ColumnMetadata(name="op_rate", unit="ops", type=ResultType.INTEGER),
+                   ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)]
+
+
+result_table = LatencyResultTable()
+
+result_table.add_result(column="latency", row="mean", value=1.1, status=Status.WARNING)
+result_table.add_result(column="op_rate", row="mean", value=59988, status=Status.ERROR)
+result_table.add_result(column="latency", row="p99", value=2.7, status=Status.PASS)
+result_table.add_result(column="op_rate", row="p99", value=59988, status=Status.WARNING)
+result_table.add_result(column="duration", row="p99", value=8888, status=Status.ERROR)
+result_table.add_result(column="duration", row="mean", value=332, status=Status.PASS)
+
+# when testing locally, otherwise just use ArgusClient instance in your test.
+run_id = UUID("24e09748-bba4-47fd-a615-bf7ea2c425eb")
+client = ArgusSCTClient(run_id, auth_token="UO+2GXL9XqSgcVJijWk5WnbPXPit5ot5nfkLAHAr7SaqROfSCWycabpp/wxyY8+I",
```

Comment:
is this a real token ?

Reply:
no, just my local development one. I revoked it and remove from here too.

Reply:
Good